src/test/playlisttest.cpp

```diff
@@ -56,3 +57,52 @@ TEST_F(PlaylistTest, Relative) {
     EXPECT_EQ(QString("base/folder/../../bar.mp3"),
             parser.playlistEntryToFilePath("../../bar.mp3", "base/folder"));
 }
+
+TEST_F(PlaylistTest, m3uEndOfLine) {
+    QTemporaryFile m3uFile;
+    ASSERT_TRUE(m3uFile.open());
+    m3uFile.write("crlf.mp3\r\n");
+    m3uFile.write("cr.mp3\r");
+    m3uFile.write("lf.mp3\n");
+    // Check for Windows-1250 Euro Sign
+    m3uFile.write("EuroSign\x80.mp3\n");
+    m3uFile.write("end.mp3");
+    m3uFile.close();
+
+    QList<QString> entries = ParserM3u().parseAllLocations(m3uFile.fileName());
+    ASSERT_EQ(entries.size(), 5);
+    EXPECT_TRUE(entries[0].endsWith(QStringLiteral("crlf.mp3")));
+    EXPECT_TRUE(entries[1].endsWith(QStringLiteral("cr.mp3")));
+    EXPECT_TRUE(entries[2].endsWith(QStringLiteral("lf.mp3")));
+    EXPECT_TRUE(entries[3].endsWith(QStringLiteral("EuroSign\u20AC.mp3")));
+    EXPECT_TRUE(entries[4].endsWith(QStringLiteral("end.mp3")));
```

Comment:
generally favor `QList.at()` over `operator[]` iirc

Reply:
Why? If the test fails, let it crash.

Reply:
.at() does not perform bounds checking. It's rather preferred because it is const and thus guarantees that the implicitly shared container will not be detached (which does not matter in this particular context but still good practice I'd argue).

Reply:
There is literally no difference: 

```
inline const T &QList<T>::operator[](int i) const
{ Q_ASSERT_X(i >= 0 && i < p.size(), "QList<T>::operator[]", "index out of range");
 return reinterpret_cast<Node *>(p.at(i))->t(); }
```
```
inline const T &QList<T>::at(int i) const
{ Q_ASSERT_X(i >= 0 && i < p.size(), "QList<T>::at", "index out of range");
 return reinterpret_cast<Node *>(p.at(i))->t(); }
```

With [] we just to be careful not to use the non const variant which is not obvious. From this point of view preferring at() is a good Idea. It is also recommended by Qt: 
> If you do not want to modify the list you should use QList::at().

Reply:
Right, I already said that it does not matter here, but we should still promote good coding standards, even in our test code which is why I brought it up.